<launch>
  <arg name="config_file" />

  <!-- Intel realsense D435i-->
  <node pkg="rs4se"
      type="intel_d435i"
      name="rs"
      required="true"
      output="screen">
    <param name="global_time" type="bool" value="true" />

    <!-- RGBD -->
    <param name="correct_ts" type="bool" value="false" />
    <param name="enable_ir" type="bool" value="true" />
    <param name="enable_rgb" type="bool" value="false" />
    <param name="enable_ir" type="bool" value="true" />
    <param name="enable_depth" type="bool" value="false" />
    <param name="enable_emitter" type="bool" value="false" />
    <param name="rgb_width" value="640"/>
    <param name="rgb_height" value="480"/>
    <param name="rgb_frame_rate" value="15"/>
    <param name="rgb_format" type="str" value="RGB8" />
    <param name="rgb_exposure" type="double" value="100" />
    <param name="ir_width" type="int" value="640" />
    <param name="ir_height" type="int" value="480" />
    <param name="ir_format" type="str" value="Y8" />
    <param name="ir_frame_rate" type="int" value="15" />
    <param name="ir_exposure" type="double" value="10000" />
    <param name="depth_width" type="int" value="640" />
    <param name="depth_height" type="int" value="480" />
    <param name="depth_format" type="str" value="Z16" />
    <param name="depth_frame_rate" type="int" value="15" />

    <!-- IMU -->
    <param name="enable_motion" type="bool" value="true" />
    <param name="gyro_hz" type="double" value="400" />
    <param name="accel_hz" type="double" value="250" />
  </node>

  <!-- RVIZ -->
  <node type="rviz" name="rviz" pkg="rviz"
    args="-d $(find yac_ros)/config/yac.rviz"
    required="true"
    output="screen" />

  <!-- NBV -->
  <node pkg="yac_ros" type="calib_nbv_node" name="yac_node" required="true" output="screen">
    <param name="config_file" value="$(arg config_file)" />
    <param name="publish_tfs" value="true" />
    <param name="image_format" value="bgr8" />
    <param name="cam0_topic" value="/rs/ir0/image" />
    <param name="cam1_topic" value="/rs/ir1/image" />
    <param name="imu0_topic" value="/rs/imu0/data" />
  </node>
</launch>
